app/models/payment.rb

```diff
@@ -40,22 +40,26 @@ def self.build(bt_customer, bt_payment_method, member_id, existing_customer)
 
     def initialize(bt_customer, bt_payment_method, member_id, existing_customer)
       @bt_customer = bt_customer
-      @bt_payment_method = bt_payment_method
       @existing_customer = existing_customer
       @member_id = member_id
+      @bt_payment_method = bt_payment_method
     end
 
     def build
       if @existing_customer.present?
         @existing_customer.update(customer_attrs)
       else
-        Payment::BraintreeCustomer.create(customer_attrs)
+        @existing_customer = Payment::BraintreeCustomer.create(customer_attrs)
       end
+      Payment::BraintreePaymentMethod.find_or_create_by!(
+        customer: @existing_customer,
+        token:    @bt_payment_method.token
+      )
+      @existing_customer.save
```

Comment:
ahh yeah I missed that. could this then go in the conditional right after the call to update? or you could just call `@existing_customer.update_attributes(customer_attrs)` and it will just do it all in one line.